lib/improver/utilities/temporal_interpolation.py

```diff
@@ -0,0 +1,419 @@
+                                         and crd.name() is not coord_name]
+                    coord_list.append(coord_name)
+                    for val in associated_coords:
+                        coord_list.append(val)
+                except CoordinateNotFoundError:
+                    msg = ('Original cube does not have the coordinate in'
+                           ' dim_coords {}'.format(coord_name))
+                    raise CoordinateNotFoundError(msg)
+        for coord_name in coord_list:
+            dtype_orig = cube.coord(coord_name).points.dtype
+            try:
+                new_coord = new_cube.coord(coord_name)
+                check_coord_datatypes(new_coord, dtype_orig, fix=True,
+                                      rounding=True)
+            except CoordinateNotFoundError:
+                msg = ('new_cube does not have the coordinate in'
+                       ' the original cube {}'.format(coord_name))
+                raise CoordinateNotFoundError(msg)
+        return new_cube
+
+    @staticmethod
+    def calc_sin_phi(dtval, lats, lons):
+        """
+        Calculate sin of solar elevation
+
+        Args:
+            dtval (datetime.datetime):
+                Date and time.
+            lats (np.array):
+                Array 2d of latitudes for each point
+            lons (np.array):
+                Array 2d of longitudes for each point
+        Returns:
+            sin_phi (np.array):
+                Array of sine of solar elevation at each point
+
+        """
+        day_of_year = (dtval - datetime(dtval.year, 1, 1)).days
+        utc_hour = (dtval.hour * 60.0 + dtval.minute) / 60.0
+        sin_phi = calc_solar_elevation(lats, lons, day_of_year,
+                                       utc_hour, return_sine=True)
+        return sin_phi
+
+    @staticmethod
+    def calc_lats_lons(cube):
+        """
+        Calculate the lats and lons of each point for a given cube.
+
+        Args:
+            cube (iris.cube.Cube):
+                cube containing x and y axis
+        Returns:
+            (tuple) : tuple containing:
+                **lats** (np.array):
+                    2d Array of latitudes for each point.
+                **lons** (np.array):
+                    2d Array of longitudes for each point.
+
+        """
+        trg_crs = lat_lon_determine(cube)
+        if trg_crs is not None:
+            xycube = next(cube.slices([cube.coord(axis='y'),
+                                       cube.coord(axis='x')]))
+            lats, lons = transform_grid_to_lat_lon(xycube)
+        else:
+            lats_row = cube.coord('latitude').points
+            lons_col = cube.coord('longitude').points
+            lats = np.repeat(lats_row[:, np.newaxis], len(lons_col), axis=1)
+            lons = np.repeat(lons_col[np.newaxis, :], len(lats_row), axis=0)
+        return lats, lons
+
+    def solar_interpolate(self, cube, interpolated_cube):
```

Comment:
Why do you need to pass in both "cube" and "interpolated_cube" here?  Doesn't "interpolated_cube" have the first and last fields at each end?

Reply:
The interpolated_cube only contains the cube  at times in the time_list i.e. not at the start and end.

Reply:
OK.